### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by prathmeshjagtap**
Added Authentication

Added Login Page and Functionality
Added Signup Page and Functionality
Used Redux 
Added Private Routes
Added Toastify to show alerts

Preview Link :  https://deploy-preview-2--habitstracker.netlify.app/

### CODE REVIEW

**Review by nerdyspook — APPROVED** [review]
Well structured, looks good to me.

File: src/frontend/screens/login/Login.jsx [hunk 1]
```diff
@@ -1,7 +1,191 @@
...
+
+			localStorage.setItem("token", response.data.encodedToken);
+			dispatch(
+				token({
+					token: response.data.encodedToken,
+					userInfo: response.data.foundUser,
+				})
+			);
+			toast.success("Logged in Successfully ", {
+				position: "top-right",
```
Comment (SimranjitKaurMaan) [inline src/frontend/screens/login/Login.jsx]: This styling can be abstracted out in a separate file.

Comment (SimranjitKaurMaan) [reply src/frontend/screens/login/Login.jsx]: You can have a styles.js and keep adding your styles there.

```
export const Tracker = {
    toastStyle : {
        position: "top-right",
       autoClose: 2000,
    },
....
}
```

And then pass the object here, 
toast.error("Wrong Email", {...Tracker.toastStyle});

**Review by SimranjitKaurMaan — APPROVED** [review]
Small suggestion. Overall clean code and understandable.

**Review by horsemaker — APPROVED** [review]
You have used Redux! Great work there! 👌
- Thanks for the motivation, might start learning redux too.
- Great Code Structure
- Understandable code
- Also, I like the use of toastify here.

File: src/frontend/screens/login/Login.jsx (hunk 1 above)
Comment (prathmeshjagtap) [reply src/frontend/screens/login/Login.jsx]: yes added this , instead of repeating  code , thanks for pointing  this out

Comment (SukanyaSen530) [reply src/frontend/screens/login/Login.jsx]: You can pass the overall style in toastcontainer. No need to pass the styles in each toast.

**Pull request merged**